### ISSUE TITLE
tuist fetch failed when Swift macro Swift package dependency added

### ISSUE THREAD

**Issue by tackgyu** [tuist/tuist#5435]
### What happened?

When Swift package containing Swift macro added, `tuist fetch` failed without error message.

### How do we reproduce it?

1. `mkdir Test`
2. `cd Test`
3. `tuist init --platform ios`
4. `echo 3.27.0 > .tuist-version`
5. touch `Tuist/Dependencies.swift`
6. Open `Tuist/Dependencies.swift` and edit as:
```swift
import ProjectDescription

let dependencies = Dependencies(
    swiftPackageManager: [
        .remote(url: "https://github.com/lukepistrol/SFSymbolsMacro", requirement: .exact("0.4.1")),
    ],
    platforms: [.iOS]
)

```
7. `tuist fetch`

### Error log

No error message emitted by tuist.

### macOS version

13.5.2

### Tuist version

3.27.0

### Xcode version

15.0

**Comment by mo5tone** [comment]
Same issue when add [MetaCodable](https://github.com/SwiftyLab/MetaCodable) as SPM dependency.

**Comment by danyf90** [comment]
This should be fixed in 3.28.0
